src/scss/src/foundation/_mixins.scss

```diff
@@ -1,16 +1,169 @@
+@use "sass:list";
 /*=============================================
 =                 #MIXINS                     =
 =============================================*/
 
-
-
 /**
  *
  * Helpful reusable functions-like code that will be used to implement
  * styles for components
  *
  */
 
+/*=============================================
+ =                @media rules                =
+ =============================================*/
+
+@mixin tablet {
+    @media (min-width: map-get($breakpoints, "tablet")) {
+        @content;
+    }
+}
+
+@mixin desktop {
+    @media (min-width: map-get($breakpoints, "desktop")) {
+        @content;
+    }
+}
+
+@mixin mobile {
+    @media (min-width: map-get($breakpoints, "mobile")) {
+        @content;
+    }
+}
+
+/*=============================================
+=                 Spacings                    =
+=============================================*/
+
+/**
+ * use it set up margin or padding 
+ * 
+ * @include spacing(margin, 'lg', 'xxs') - will set up margin vertical for 32px and horizontal for 8px 
+ */
+
+@mixin spacing($property, $values...) {
+    $valid-properties: margin, padding;
+
+    @if not list.index($valid-properties, $property) {
+        @error "#{$property} is not a valid property. Expected one of #{$valid-properties}.";
+    }
+
+    $propertyValues: ();
 
+    @each $value in $values {
+        $spacingValue: map-get($spacing, $value);
+        $propertyValues: append($propertyValues, $spacingValue);
+    }
 
+    #{$property}: $propertyValues;
+}
+
+/*=============================================
+=              Containers sizings            =
+=============================================*/
+
+/**
+ * defines width and height in one go
+ * 
+ * @include box(1rem, 2rem) => width: 1rem;  height: 2rem
+ */
+
+@mixin box($width, $height: $width) {
+    width: $width;
+    height: $height;
+}
+
+/*=============================================
+=               FLEX/GRID Toolkit            =
+=============================================*/
+
+/**
+ * Setup for flexbox container
+ *
+ * example usage: 
+ * @include flexbox(column, space-between)
+ * @include flexbox($justify: center)
+ *
+ */
+
+@mixin flexbox($direction: row, $justify: center, $align: center, $wrap: nowrap, $gap: normal) {
+    display: flex;
+    flex-direction: $direction;
+    justify-content: $justify;
+    align-items: $align;
+    flex-wrap: $wrap;
+    gap: $gap;
+}
+
```

Comment:
Does this mixin and others provide default values for unused arguments or properties are added with an error?

Reply:
All values which are followed by the semicolons are default values
F.e `$direction: row` - if not provided will be set to `row`
There is an example hwo we can use this mixin, either like a standart js function or manually ype only those arguments which we want to set